api/simplehash/schemav2.go

```diff
@@ -0,0 +1,198 @@
+		opt(&o)
+	}
+
+	var err error
+
+	// By default, one hash at at time with a reset.
+	if !o.accumulateHash {
+		h.hasher.Reset()
+	}
+
+	if o.publicFromPermissioned {
+		PublicFromPermissionedEvent(event)
+	}
+
+	// If the caller is responsible for evidence confirmation they will have a
+	// pending event in their hand. But ultimately it is the confirmed record
+	// that is evidential and subject to public verification.
+	if o.asConfirmed {
+		event.ConfirmationStatus = v2assets.ConfirmationStatus_CONFIRMED
+	}
+	err = EventSimpleHashV2(h.hasher, h.marshaler, event)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *HasherV2) Sum() []byte {
+	return h.hasher.Sum(nil)
+}
+
+// EventV2 is a struct that contains ONLY the event fields we want to hash for schema v2
+type EventV2 struct {
+	Identity           string         `json:"identity"`
+	AssetIdentity      string         `json:"asset_identity"`
+	EventAttributes    map[string]any `json:"event_attributes"`
+	AssetAttributes    map[string]any `json:"asset_attributes"`
+	Operation          string         `json:"operation"`
+	Behaviour          string         `json:"behaviour"`
+	TimestampDeclared  string         `json:"timestamp_declared"`
+	TimestampAccepted  string         `json:"timestamp_accepted"`
+	TimestampCommitted string         `json:"timestamp_committed"`
+	PrincipalAccepted  map[string]any `json:"principal_accepted"`
+	PrincipalDeclared  map[string]any `json:"principal_declared"`
+	ConfirmationStatus string         `json:"confirmation_status"`
+	From               string         `json:"from"`
+	TenantIdentity     string         `json:"tenant_identity"`
+}
+
+// NewEventMarshaler creates a flat marshaler to transform events to api format.
+//
+// otherwise attributes look like this: {"foo":{"str_val": "bar"}} instead of {"foo": "bar"}
+// this mimics the public list events api response, so minimises changes to the
+// public api response, to reproduce the anchor
+func NewEventMarshaler() *simpleoneof.Marshaler {
+	return v2assets.NewFlatMarshalerForEvents()
+}
+
+// TransformOne transforms a single event in grpc proto format (message bus
+// compatible) to the canonical, publicly verifiable, api format.
+func TransformOne(marshaler *simpleoneof.Marshaler, event *v2assets.EventResponse) (EventV2, error) {
+	eventJson, err := marshaler.Marshal(event)
+	if err != nil {
+		return EventV2{}, err
+	}
+	eventShashV2 := EventV2{}
+	err = json.Unmarshal(eventJson, &eventShashV2)
+	if err != nil {
+		return EventV2{}, err
+	}
+	return eventShashV2, nil
+}
+
+// PublicFromPermissionedEvent translates the permissioned event and asset identities to
+// their public counter parts.
+func PublicFromPermissionedEvent(event *v2assets.EventResponse) {
+	event.Identity = v2assets.PublicIdentityFromPermissioned(event.Identity)
+	event.AssetIdentity = v2assets.PublicIdentityFromPermissioned(event.AssetIdentity)
+}
+
+// EventSimpleHashV2 hashes a single event according to the public event format
```

Comment:
minor: what is meant by `public event format`? 

I maybe confused as a consumer into reading only public events are allowed to be hashed, but that is not the case

Reply:
ah yeah, I will avoid that wording ta